src/DataFactory.js

```diff
@@ -85,6 +85,9 @@ class DataFactory {
         }
 
         if (type.name === 'list') {
+            if (!Array.isArray(value)) {
```

Comment:
What's the reason to not validate it in the type class ? DataFactory should be general

Reply:
Because later we are calling the `map` method of `value`, should we move it to `FateList` as well?

Reply:
Fair point, I've overlooked that.

However there is non-sense tautology in the error message "list must be an array", same for tuple: "Tuple must be an array". Please try to find a better error message.

Something like "[Fate] List constructor expects an array of values, got X instead"

And slight note on formatting, please have a new line after each block of statements, if that not the last statement in another block. I know I've to commit the rules and it's not obvious from the codebase .. soon

Reply:
> [Fate] List constructor expects an array of values, got X instead

I'm not sure about the exact message
- won't "Sophia" be more clear than "Fate" for third-party users?
- the constructor seems to be too internal part to mention it
- I would drop "of values" because this is the same for any array, technically we can generate the exact type instead (like array of numbers)
- would be better to name Array from the capital letter to highlight that this is a JS type?
- should we do `JSON.stringify` of X? should we handle the "Do not know how to serialize a BigInt" exception then?

Proposing to change it to "Sophia list must be an Array, got X instead"

Reply:
My message was just an example, feel free to change it to whatever feel better. However, again on users .. this library has nothing to do with Sophia programming language, but FATE virtual machine :) So please don't use the term sophia, you don't have to use the term fate as well if you don't think it's a good fit

Reply:
What if later we have multiple backends (as aevm and fate before) would be correct to use Sophia term or we would name it by backend even if input should have the same type for both vms? My idea is that this library at the end should convert Sophia arguments dispite the backend that is currently used. User can pass the calldata to blockchain without even knowing what backend is used 🤔

Reply:
I follow the YAGNI principle, and this lib is about FATE only, other backend would probably mean additional lib

Reply:
@dincho should we think about renaming the repo then better sooner than later? or would you propose to just call the separate lib sth. like `aepp-calldata-evm-js`?

Reply:
also when looking into the readme that might be confusing. just want to raise that topic now :-)

Reply:
I don’t know really, AEVM is deprecated, so the only VM now is FATE. We also don’t if there would be any other VM in future so let’s keep it simple? I’m more lean to drop off the calldata from the name as I plan to add full serializaion support anytime soon, however we might just split the serializer and encoder 🤔🤷‍♂️

Reply:
> also when looking into the readme that might be confusing. just want to raise that topic now :-)

Would you propose readme changes?

Reply:
I shouldn't have interferred here 😅 - actually I currently don't have concrete proposals. I am fine with leaving like is for the moment in regards to the readme. 

> We also don’t know if there would be any other VM in future so let’s keep it simple?

👍 

> I’m more lean to drop off the calldata from the name as I plan to add full serializaion support anytime soon

might make sense in that case to do this better earlier than later. we should have some reliable release in the near future 🤔

Reply:
1.0.0-rc1 perhaps